std/experimental/allocator/building_blocks/allocator_list.d

```diff
@@ -173,8 +173,8 @@ struct AllocatorList(Factory, BookkeepingAllocator = GCAllocator)
                 *p = n.next;
                 n.next = root;
                 root = n;
-                return result;
             }
+            return result;
```

Comment:
I added a unittest. It checks the value of private field. I don't know whether to do such checks in unittests, but I have no idea how to check it otherwise.